tasks/supportuser.sh

```diff
@@ -2,27 +2,14 @@
 
 # Puppet Task Name: supportuser
 #
-# This is where you put the shell code for your task.
-#
-# You can write Puppet tasks in any language you want and it's easy to
-# adapt an existing Python, PowerShell, Ruby, etc. script. Learn more at:
-# https://puppet.com/docs/bolt/0.x/writing_tasks.html
-#
-# Puppet tasks make it easy for you to enable others to use your script. Tasks
-# describe what it does, explains parameters and which are required or optional,
-# as well as validates parameter type. For examples, if parameter "instances"
-# must be an integer and the optional "datacenter" parameter must be one of
-# portland, sydney, belfast or singapore then the .json file
-# would include:
-#   "parameters": {
-#     "instances": {
-#       "description": "Number of instances to create",
-#       "type": "Integer"
-#     },
-#     "datacenter": {
-#       "description": "Datacenter where instances will be created",
-#       "type": "Enum[portland, sydney, belfast, singapore]"
-#     }
-#   }
-# Learn more at: https://puppet.com/docs/bolt/0.x/writing_tasks.html#ariaid-title11
-#
+password=$(/opt/puppetlabs/puppet/bin/openssl rand -base64 32) #Set Random Password for Support Client tools account
+
+curl -X POST -H 'Content-Type: application/json' --cert $(puppet config print hostcert) --key $(puppet config print hostprivkey) --cacert $(puppet config print localcacert) https://$(hostname -f):4433/rbac-api/v1/users -d "{\"login\":\"pesupport\",\"email\":\"support@puppet.com\",\"role_ids\": [],\"display_name\":\"Puppet Enterprise Support\", \"password\": \"$password\"}"
+
+echo "password for pesupport account is $password"
```

Comment:
@MartyEwings Do we intend to to use the output of this in a plan at all, or is it purely informational? If it's a plan then I'd recommend formatting it as JSON so that bolt does the parsing for us i.e.

```shell
echo '{"status":"created","password":"$password"}'
```

That way it will be parsed and returned as a Puppet data structure natively

Reply:
its not going to be used in a plan, but it couldn't hurt to future proof, ill make the changes

Reply:
![Screenshot 2021-01-04 at 14 53 06](https://user-images.githubusercontent.com/26789261/103547701-ab419a00-4e9c-11eb-9904-82adc9e50a92.png)
tested and working

Reply:
It'll need literal double quotes in the output to be valid JSON, like around each string. You'll probably have to escape the double quotes so that they actually end up in the output